<?xml version="1.0"?>
<launch>
    <include file="$(find description)/launch/reflex.launch">
        <arg name="gui" value="false" />
    </include>

    <arg name="agent_spawn_delay" default="10" />

    <!-- args for training script -->
    <arg name="environment" default="refinement" />
    <arg name="train" default="1" />
    <arg name="seed" default="0" />
    <arg name="max_ep_len" default="15" />
    <arg name="joint_lim" default="2" />
    <arg name="exec_secs" default="0.3" />
    <arg name="obj_shift_tol" default="0.03" />
    <arg name="time_steps" default="2000" />
    <arg name="reward_weight" default="1" />
    <arg name="log_name" default="test" />
    <arg name="output_dir" default="./" />
    <arg name="chkpt_freq" default="300" />
    <arg name="x_error_min" default="0" />
    <arg name="y_error_min" default="0" />
    <arg name="z_error_min" default="0" />
    <arg name="x_error_max" default="0" />
    <arg name="y_error_max" default="0" />
    <arg name="z_error_max" default="0" />
    <arg name="framework" default="1" />
    
    <node pkg="agent" name="agent" type="main.py" output="screen" launch-prefix="bash -c 'sleep $(arg agent_spawn_delay); $0 $@' " 
    args="
        --environment=$(arg environment) 
        --train=$(arg train) 
        --seed=$(arg seed) 
        --max_ep_len=$(arg max_ep_len) 
        --joint_lim=$(arg joint_lim) 
        --exec_secs=$(arg exec_secs) 
        --obj_shift_tol=$(arg obj_shift_tol) 
        --time_steps=$(arg time_steps) 
        --reward_weight=$(arg reward_weight) 
        --log_name=$(arg log_name) 
        --output_dir=$(arg output_dir) 
        --chkpt_freq=$(arg chkpt_freq)
        --x_error_min=$(arg x_error_min) 
        --y_error_min=$(arg y_error_min) 
        --z_error_min=$(arg z_error_min) 
        --x_error_max=$(arg x_error_max) 
        --y_error_max=$(arg y_error_max) 
        --z_error_max=$(arg z_error_max) 
        --framework=$(arg framework)"/>
</launch>
